server/cluster/cluster.go

```diff
@@ -254,6 +254,98 @@ func (c *Cluster) updateTableCacheLocked(shardID uint32, schema *Schema, tablePb
 	return table
 }
 
+func (c *Cluster) updateShardTopologyVersionLocked(shardID uint32, storageVersion uint64) (*ShardVersion, error) {
+	shard, ok := c.shardsCache[shardID]
+	if !ok {
+		return nil, ErrShardNotFound
+	}
+
+	prevVersion := shard.version
+	if prevVersion != storageVersion {
+		panic(fmt.Sprintf("shardId:%d, storage version:%d, memory version:%d", shardID, storageVersion, prevVersion))
+	}
+	shard.version = prevVersion + 1
+
+	return &ShardVersion{
+		ShardID:     shardID,
+		CurrVersion: shard.GetVersion(),
+		PrevVersion: prevVersion,
+	}, nil
+}
+
+func (c *Cluster) createTableUpdateTopologyLocked(ctx context.Context, shardID uint32, schema *Schema, tablePb *clusterpb.Table) (*CreateTableResult, error) {
+	// Update shardTopology in storage.
+	shardTopologies, err := c.storage.ListShardTopologies(ctx, c.clusterID, []uint32{shardID})
+	if err != nil {
+		return nil, errors.WithMessage(err, "get shard topology from storage")
+	}
+	if len(shardTopologies) != 1 {
```

Comment:
It's weird to check length of shardTopologies, it's better to define `GetShardTopologyByID` and replace `ListShardTopologies`.